x-pack/plugins/security/server/lib/watch_status_and_license_to_initialize.js

```diff
@@ -4,23 +4,31 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import * as Rx from 'rxjs';
-import { catchError, mergeMap, map, retryWhen, switchMap, tap } from 'rxjs/operators';
+import { catchError, mergeMap, map, switchMap, tap } from 'rxjs/operators';
 
-export const retryStrategy = ({
-  maxAttempts,
-  scalingDuration,
-}) => (errors) => {
-  return errors.pipe(
-    mergeMap((error, i) => {
-      const attempt = i + 1;
+export const RETRY_SCALE_DURATION = 100;
+export const RETRY_DURATION_MAX = 10000;
 
-      if (attempt >= maxAttempts) {
-        return Rx.throwError(error);
-      }
+const calculateDuration = i => {
+  const duration = i * RETRY_SCALE_DURATION;
+  if (duration > RETRY_DURATION_MAX) {
+    return RETRY_DURATION_MAX;
+  }
 
-      return Rx.timer(attempt * scalingDuration);
-    })
-  );
+  return duration;
+};
+
+// we can't use a retryWhen here, because we want to propagate the red status and then retry
+const propagateRedStatusAndScaleRetry = () => {
+  let i = 0;
```

Comment:
I realize this counter is safe because `propagateRedStatusAndScaleRetry()` is called each time the status or license is updated, but I wonder if there's a way we could implement this so that this implicit dependency wasn't there...

Reply:
I thought about using a Rx.interval to continually emit numbers that were used to pre-determine the delay to drive this process, but it ended up looking a lot "uglier" so I reverted to this side-effect driven approach.

I also played around with using [repeatWhen](https://rxjs-dev.firebaseapp.com/api/operators/repeatWhen) after mapping a success to the green status, and an error to the red status, but that didn't work because the repeatWhen notifications observable didn't actually get the values that were being emitted... so I kinda gave up and settled on this.

If you have any other ideas for how to structure this, I'm all ears!